week_2/pet.rb

```diff
@@ -1,115 +1,140 @@
-  # as input and returns the cost to feed them all 
-  # for the specified number of `days`
-  #
-  # Eg -
-  # cat = Pet.new(name: 'cat', animal_type_id: 6, food_consumed_per_day: 0.4)
-  # dog = Pet.new(name: 'dog', animal_type_id: 6, food_consumed_per_day: 0.7)
-  # fish = Pet.new(name: 'clownfish', animal_type_id: 2, food_consumed_per_day: 0.1)
-  # snake = Pet.new(name: 'python', animal_type_id: 4, food_consumed_per_day: 0.3)
-  # Pet.cost_to_feed([cat, dog, fish, snake], 6) will return 6180.0
-  def self.cost_to_feed(pets, days)
-    raise NotImplementedError # TODO
-  end
-
-  # This function takes an array of pets as input
-  # and returns a hash with pets name grouped by their animal type
-  #
-  # Eg - 
-  # cat = Pet.new(name: 'cat', animal_type_id: 6, food_consumed_per_day: 0.4)
-  # dog = Pet.new(name: 'dog', animal_type_id: 6, food_consumed_per_day: 0.7)
-  # fish = Pet.new(name: 'clownfish', animal_type_id: 2, food_consumed_per_day: 0.1)
-  # snake = Pet.new(name: 'python', animal_type_id: 4, food_consumed_per_day: 0.3)
-  # Pet.group_by_animal_type([cat, dog, fish, snake]) will return the follwing hash
-  # {
-  #   6 => ['cat', 'dog'],
-  #   2 => ['clownfish'],
-  #   4 => ['python']
-  # }
-  #
-  # Note - Order is not important
-  def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
-  end
-end
+class Pet
+  ANIMAL_TYPE = {
+    1 => 'Arthropod',
+    2 => 'Fish',
+    3 => 'Amphibia',
+    4 => 'Reptiles',
+    5 => 'Birds',
+    6 => 'Mammals'
+  }
+
+  # Each animal type requires a diffrent type of food
+  # The food cost per 1 kg is is stored in the `FOOD_COST_PER_KG` Hash
+  # where the `key` is the animal id and `value` is the cost
+  #
+  # Eg- The food of reptiles(animal_type id = 4) costs 300/kg.
+  # (FOOD_COST_PER_KG[4] = 300)
+  FOOD_COST_PER_KG = {
+    1 => 100,
+    2 => 600,
+    3 => 1000,
+    4 => 300,
+    5 => 450,
+    6 => 800
+  }
+
+  # Each animal requires its own habitat.
+  #
+  # The `HABITATS` hash stores the habitat required by each animal
+  # The `key` is the habitat and the `value` is the array of
+  # animal_type ids that live in that habitat.
+  #
+  # "habitat" => [animal_type_id]
+  HABITATS = {
+    "Aquatic" => [2, 1, 3],
+    "Terrariums" => [4],
+    "Cages" => [5, 6]
+  }
+
+  attr_accessor :name, :animal_type_id, :food_consumed_per_day
+
+  def initialize(name: , animal_type_id: , food_consumed_per_day:)
+    @name = name
+    @animal_type_id = animal_type_id
+    @food_consumed_per_day = food_consumed_per_day
+  end
+
+  # Return the habitat of the pet
+  def habitat
+    HABITATS["Aquatic"].each do |i|
+      if self.animal_type_id == i
+        return "Aquatic"
+      end
+    end
+    HABITATS["Terrariums"].each do |i|
+      if self.animal_type_id == i
+        return "Terrariums"
+      end
+    end
+    HABITATS["Cages"].each do |i|
+      if self.animal_type_id == i
+        return "Cages"
+      end
+    end

```

Comment:
If a new HABITAT is to be added, here, both the HABITATS hash and this function has to be updated, which is not good. You can `each` to iterate over a hash too.

Reply:
Thank you for the suggestion. I updated the habitats function 
![Screenshot 2023-01-03 142859](https://user-images.githubusercontent.com/99561021/210326302-f6d5bdd3-d5d2-4391-9b18-c584e7ff7468.jpg)